### ISSUE TITLE
Companion App crashes continually

### ISSUE THREAD

**Issue by clydebarrow** [home-assistant/iOS#2810]
**iOS device model, version and app version**

Model Name: iPhone 11 Pro
Software Version: 17.5.1
App version: 2024.5.1

**Home Assistant Core Version**

2024.6.1
**Describe the bug**

The app crashes shortly after opening, sometimes immediately, sometimes after short duration.

**To Reproduce**
Open the app, either just wait, or navigate the UI.
**Expected behavior**
App should not crash.
**Screenshots**

Screen recording here. https://drive.google.com/file/d/1Rm42dAqvo-aK6fPoXze0_qhF4pkib_1L/view?usp=drivesdk

**Additional context**

I have tried deleting the app and reinstalling, no change in behaviour. Access to the same HA instance via browser on same device works without problem. Access via Android app to same instance works mostly ok (no crashing)

**Comment by bgoncal** [comment]
Some things you could try:
- Delete te App
- Reboot iPhone (Important step)
- Install the app again

If that doesn't work:
- Install out testflight app version and see if it fixes the issue, otherwise use testflight to install older version of the app (one that you know it was not crashing) and let me know which one.

Meanwhile I am checking the logs to see if I find your crash, do you know if you have crash reports enabled in iOS? (Usually you decide this when setup the iPhone itself), check in Settings >> Privacy >> Analytics & Improvements

  
![IMG_709DF13159CB-1](https://github.com/home-assistant/iOS/assets/5808343/32d57d86-db7d-4273-a2b9-2bf0cb5bf82b)

**Comment by erkr** [comment]
I experience many crashes per hour as well. Especially while editing Lovelace cards.

**Comment by erkr** [comment]
The testflight version crashes quite often as well

**Comment by erkr** [comment]
For some reason there is no option to share crash data with app developers on my iPhone 14 - iOS 17.5.1:

![image](https://github.com/home-assistant/iOS/assets/4363599/512d08ad-9cdf-48f0-bf74-603638031b2f)

**Comment by bgoncal** [comment]
Is this still happening?

**Comment by erkr** [comment]
I installed the latest test flight  version yesterday. Will keep an eye on it

**Comment by erkr** [comment]
Still happens but Less frequent, around once twice per hour it reloads. Recoding is therefore not an option

**Comment by bgoncal** [comment]
Please after 2024.8 core releases, let me know how it behaves to you

**Comment by ChuckRamsey** [comment]
Mine keeps crashing all the time. Virtually unuseable. Also gat ‘connection list. Reconnecting …’ every few seconds. Can i get you a lig somewhere and if so how gan i get it?

**Comment by bgoncal** [comment]
> Mine keeps crashing all the time. Virtually unuseable. Also get ‘connection lost. Reconnecting …’ every few seconds. Can i get you a log somewhere and if so how can i get it?

https://docs.google.com/forms/d/1lx0hKiN8D-YDTmMKybVeiBlrj4S4s1hczLrgNXTZsW8/edit

**Comment by bgoncal** [comment]
Also send your logs please: https://forms.gle/TPe6Bmn39LNuHy2M7

(companion app settings >> debugging >> export logs)

**Comment by tenallero** [comment]
Hi,

I have several devices
iPad Pro (2nd gene) with iOS 16.7.10
iPad (5th gen) with iOS 16.7.10
iPhone 11 with iOS 17
iPhone 12 with iOS 18

My Home Assistant Core release is 2024.10.1
I use Nabucasa Cloud for external URL

Since june/july (approximately) HomeAssistant companion has become much slower. And on older devices (iPad & iPadPro) it crashes after a few seconds. I don't know if this is due to companion or iOS upgrades.

Regarding older devices (iPads) I have tried everything. Delete the app with data, reset the device to factory defaults. Run with TestFlight to the oldest release (2024.7)
The results are always the same. Once the app has connected to HA ... the app freezes to crash. I can't even use the app settings options, obtain logs, etc.

Note:
For all my devices, accesing to HA via browser does not have any slowness or crashing

**Comment by cammurray** [comment]
Happening for me now, iOS 18.2 beta 2 - app loads, can sometimes navigate to one page, sometimes it doesn’t even get that far, then crash.

**Comment by hapklaar** [comment]
For me the app also crashes regularly, but only after a day of running or so. I use it as a permanent-on device (iPad with iOS 18.1.1)

Most of the time in the form of a black screen, sometimes with the yellow cogwheel on the bottom. Always unresponsive. I have to force quit the app and restart to get it going again.

This has been an issue for a couple of months at least, not sure when it started unfortunately. Please let me know how I can help debug this issue.

**Comment by bgoncal** [comment]
@tenallero try the latest testflight build and let me know if it helps

**Comment by tenallero** [comment]
@bgoncal 

I have just installed and tested the latest testflight release
For the iPad (5th gen) and iPad Pro (2nd. gen) ... it crashes after showing frontend or accesing to setup
For both devices, accessing HA frontend using safari works perfectly.

For my iPhone 12 ... the companion app does not crash. But it is very slow.
Accesing the frontend via browser ... I have no problem and much faster that companion app

**Comment by hapklaar** [comment]
@bgoncal I don't want to hijack this issue with mine as my app doesn't really crash but freezes. Should I create a separate ticket?

**Comment by tenallero** [comment]
@bgoncal 
> @tenallero can you reach the export logs button? if, submit those logs to me https://forms.gle/Uoqz127Phx4mMTpS6
Hi. Imposible to reach expot log files with both iPad devices. I have sent log files from an iPhone12 connected to the sema HA server. Maybe you found something related to my setup

**Comment by ChuckRamsey** [comment]
I am currently on 2024.1006 on an iPhone 13. Had crashes and freezes. Last weekend upgraded from Pi4 to Pi5. Apart from a dramatic increase in speed the Companion App now works like a charm. No crashes, no freezes. Also no more 'Connection lost. Reconnecting' anymore. So maybe, just maybe crashing and freezing has to do with an overloaded server not responding in time?

**Comment by tenallero** [comment]
I know it´s a different app. But, perhaps it provides some kind of information.
With the same HA server reported previously, I have a HA Companion for Mac (2024.1006) running in my Mac Studio M1 Max.
My HA companion for Mac consumes 90-100% of CPU constinously.

Since this "HA companion" performance/crash problem ... I run HA frontend using browser (safari, chrome) from all my devices without any problem. With my old iPad, my old iPhones 11, with Mac Studio. From inside or outside the house.

HA server is virtualized on a Proxmox server. Consuming an average of 15% CPU and 40% of RAM

**Comment by bgoncal** [comment]
@tenallero have you tested the latest testflight beta on your mac as well?

**Comment by bgoncal** [comment]
Looking at your database, it should be related to your 10k entities updating and notifying the app of it's new state, thats my current theory. Do you have entities which update all the time, like many times per second?

**Comment by tenallero** [comment]
Yes,
In the latest year I have included a lot of new entities that update every second (voltages, amperages, energy measurement)

 
> Looking at your database, it should be related to your 10k entities updating and notifying the app of it's new state, thats my current theory. Do you have entities which update all the time, like many times per second?

**Comment by bgoncal** [comment]
@tenallero drop me a message in discord if you can (bgoncal2)

**Comment by tenallero** [comment]
@bgoncal 
Sorry Bruno, I didn´t have time to set up a discord account to ping to you.
But .... I have installed the latest release **2024.1025** using testFlight. HA Core  also upgraded to 2024.12.1
And .... it works much much much better in my devices (including old iPad)
No freezing and still no crashing detected. Very fast initial page loading and transitioning between panels.
My old devices has returned to life. Good job and thanks!
